asset/asset.go

```diff
@@ -8,4 +8,7 @@ import (
 //go:embed *.html
 var files embed.FS
 
-var ShareTemplate = template.Must(template.ParseFS(files, "share.html"))
+var (
+	ShareTemplate = template.Must(template.ParseFS(files, "share.html"))
+	IndexTemplate = template.Must(template.ParseFS(files, "index.html"))
```

Comment:
As the index.html isn't really a template, you can just import it as byte array, then there is no need for the error handling:

asset.go
```go
//go:embed share.html
var files embed.FS

//go:embed index.html
var IndexHTML []byte

var (
	ShareTemplate = template.Must(template.ParseFS(files, "share.html"))
)
```
home.go
```go
func (h *Handler) Home(rw http.ResponseWriter, r *http.Request) {
	rw.Header().Set("content-type", "text/html")
    if _, err := rw.Write(asset.IndexHTML); err != nil {
        // log error to h.log
    }
}
```